app/controllers/questions_controller.rb

```diff
@@ -0,0 +1,57 @@
+class QuestionsController < ApplicationController
+  before_action :set_question, only: [:edit, :update, :destroy, :show]
+
+  def new
+    @question = Question.new
+  end
+
+  def edit
+  end
+
+  def create
+    @question = current_user.questions.build(question_params)
+    respond_to do |format|
+      if @question.publish_question(params[:publish])
```

Comment:
create should not publish directly and even if that's required use attr_accessor isntead of passing params
let's discuss